translate-c: scoping issues on Windows

### Zig Version

0.10.0-dev.3736+7453f56e6

### Steps to Reproduce

1. download windows dev kit
2. `dev-kit\zig.exe build -p stage3 -Dstatic-llvm -Denable-stage1 --search-prefix dev-kit --zig-lib-dir lib -Duse-zig-libcxx -Dtarget=x86_64-windows-gnu`
3. `stage3\bin\zig.exe build test-run-translated-c -Dstatic-llvm -Dskip-non-native --search-prefix dev-kit`

### Expected Behavior

All tests pass

### Actual Behavior

```
C:\Users\kubkon\dev\zig\stage3\lib\zig\std\log.zig:139:17: error: expected 1 argument(s), found 4
            root.log(message_level, scope, format, args);
            ~~~~^~~~
C:\Users\kubkon\AppData\Local\zig\o\20a4f45623f3abcbca2fb05762a165d8\source.zig:158:12: note: function declared here
pub extern fn log(_X: f64) f64;
~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
error: run-translated-c NAN and INFINITY build-exe...
error: The following command exited with error code 1:
C:\Users\kubkon\dev\zig\stage3\bin\zig.exe build-exe C:\Users\kubkon\AppData\Local\zig\o\20a4f45623f3abcbca2fb05762a165d8\source.zig -lc --cache-dir C:\Users\kubkon\dev\zig\zig-cache --global-cache-dir C:\Users\kubkon\AppData\Local\zig --name translated_c -L C:\Users\kubkon\dev\zig-llvm\lib -isystem C:\Users\kubkon\dev\zig-llvm\include --enable-cache
error: the following build command failed with exit code 1:
C:\Users\kubkon\dev\zig\zig-cache\o\9f546ce174b20e882b94b3b9a6171c2f\build.exe C:\Users\kubkon\dev\zig\stage3\bin\zig.exe C:\Users\kubkon\dev\zig C:\Users\kubkon\dev\zig\zig-cache C:\Users\kubkon\AppData\Local\zig test docs -Dstatic-llvm -Dskip-non-native --search-prefix C:\Users\kubkon\dev\zig-llvm -Dskip-stage2-tests
```

--- a/lib/std/coff.zig
+++ b/lib/std/coff.zig
@@ -841,10 +841,7 @@ pub const Coff = struct {
         if (self.is_image) {
             self.coff_header_offset = coff_header_offset + 4;
             const coff_header = self.getCoffHeader();
-            if (coff_header.size_of_optional_header == 0) {
-                std.log.err("Required PE header missing for image file", .{});
-                return error.MissingPEHeader;
-            }
+            if (coff_header.size_of_optional_header == 0) return error.MissingPEHeader;
         }
 
         // JK: we used to check for architecture here and throw an error if not x86 or derivative.

--- a/test/run_translated_c.zig
+++ b/test/run_translated_c.zig
@@ -1,5 +1,4 @@
 const std = @import("std");
-const builtin = @import("builtin");
 const tests = @import("tests.zig");
 const nl = std.cstr.line_sep;
 
@@ -1684,41 +1683,38 @@ pub fn addCases(cases: *tests.RunTranslatedCContext) void {
     // TODO: add isnan check for long double once bitfield support is added
     //       (needed for x86_64-windows-gnu)
     // TODO: add isinf check for long double once std.math.isInf supports c_longdouble
-    // TODO https://github.com/ziglang/zig/issues/12630
-    if (!(builtin.zig_backend == .stage2_llvm and builtin.target.os.tag == .windows)) {
-        cases.add("NAN and INFINITY",
-            \\#include <math.h>
-            \\#include <stdint.h>
-            \\#include <stdlib.h>
-            \\union uf { uint32_t u; float f; };
-            \\#define CHECK_NAN(STR, VAL) { \
-            \\    union uf unpack = {.f = __builtin_nanf(STR)}; \
-            \\    if (!isnan(unpack.f)) abort(); \
-            \\    if (unpack.u != VAL) abort(); \
-            \\}
-            \\int main(void) {
-            \\    float f_nan = NAN;
-            \\    if (!isnan(f_nan)) abort();
-            \\    double d_nan = NAN;
-            \\    if (!isnan(d_nan)) abort();
-            \\    CHECK_NAN("0", 0x7FC00000);
-            \\    CHECK_NAN("", 0x7FC00000);
-            \\    CHECK_NAN("1", 0x7FC00001);
-            \\    CHECK_NAN("0x7FC00000", 0x7FC00000);
-            \\    CHECK_NAN("0x7FC0000F", 0x7FC0000F);
-            \\    CHECK_NAN("0x7FC000F0", 0x7FC000F0);
-            \\    CHECK_NAN("0x7FC00F00", 0x7FC00F00);
-            \\    CHECK_NAN("0x7FC0F000", 0x7FC0F000);
-            \\    CHECK_NAN("0x7FCF0000", 0x7FCF0000);
-            \\    CHECK_NAN("0xFFFFFFFF", 0x7FFFFFFF);
-            \\    float f_inf = INFINITY;
-            \\    if (!isinf(f_inf)) abort();
-            \\    double d_inf = INFINITY;
-            \\    if (!isinf(d_inf)) abort();
-            \\    return 0;
-            \\}
-        , "");
-    }
+    cases.add("NAN and INFINITY",
+        \\#include <math.h>
+        \\#include <stdint.h>
+        \\#include <stdlib.h>
+        \\union uf { uint32_t u; float f; };
+        \\#define CHECK_NAN(STR, VAL) { \
+        \\    union uf unpack = {.f = __builtin_nanf(STR)}; \
+        \\    if (!isnan(unpack.f)) abort(); \
+        \\    if (unpack.u != VAL) abort(); \
+        \\}
+        \\int main(void) {
+        \\    float f_nan = NAN;
+        \\    if (!isnan(f_nan)) abort();
+        \\    double d_nan = NAN;
+        \\    if (!isnan(d_nan)) abort();
+        \\    CHECK_NAN("0", 0x7FC00000);
+        \\    CHECK_NAN("", 0x7FC00000);
+        \\    CHECK_NAN("1", 0x7FC00001);
+        \\    CHECK_NAN("0x7FC00000", 0x7FC00000);
+        \\    CHECK_NAN("0x7FC0000F", 0x7FC0000F);
+        \\    CHECK_NAN("0x7FC000F0", 0x7FC000F0);
+        \\    CHECK_NAN("0x7FC00F00", 0x7FC00F00);
+        \\    CHECK_NAN("0x7FC0F000", 0x7FC0F000);
+        \\    CHECK_NAN("0x7FCF0000", 0x7FCF0000);
+        \\    CHECK_NAN("0xFFFFFFFF", 0x7FFFFFFF);
+        \\    float f_inf = INFINITY;
+        \\    if (!isinf(f_inf)) abort();
+        \\    double d_inf = INFINITY;
+        \\    if (!isinf(d_inf)) abort();
+        \\    return 0;
+        \\}
+    , "");
 
     cases.add("signed array subscript. Issue #8556",
         \\#include <stdint.h>

